--- SublimeCodeIntel.py	2013-09-13 16:17:45.000000000 +0200
+++ SublimeCodeIntel.py	2013-09-13 16:21:45.000000000 +0200
@@ -341,14 +341,21 @@
                             params = [p.strip() for p in m.group(2).split(',')]
                             if params:
                                 snippet = []
+                                _completions = []
                                 for i, p in enumerate(params):
                                     if p:
+                                        p = p.strip()
+                                        argname = p.split('=')[0]
+                                        _completions.append((argname + ' (argument)', argname))
+                                        if p.find('=') != -1:
+                                            continue
                                         var, _, _ = p.partition('=')
                                         if ' ' in var:
                                             var = var.split(' ')[1]
                                         if var[0] == '$':
                                             var = var[1:]
                                         snippet.append('${%s:%s}' % (i + 1, var))
+                                completions[vid] = _completions
                                 contents = ', '.join(snippet)
                                 # func = m.group(1)
                                 # scope = view.scope_name(pos)
